### PR TITLE
Adds basics article on Barriers

### PR DESCRIPTION
**Pull request by StylingAndroid**
Barriers are a very wonderful thing. Let's share the barrier love...

### CODE REVIEW

File: basics/barriers.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+cl_version: 1.1.0-beta1
+order: 5
+---
+### Barriers
+
+When we are creating Android layouts, sometimes we encounter situations where the layout can change depending on the localisation. Here is a really simple example:
+
+![Aligned Layout (English)](../assets/images/basics/barriers_alignment_en.png)
+
+Here we have three TextViews: `textView1` and `textView2` on the left; and `textView3` on the right. `textView3` has been constrained to the end of `textView1` and this works perfectly - it positions and sizes`textView3` exactly as we need it.
```
Comment (welsinga) [inline basics/barriers.md]: missing space between sizes and`textView3`

File: basics/barriers.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+ 
+ However, things become more complicated if we need to support multiple languages. If we add a German translation then we have a problem because in the English version the text in `textView1` is longer than that in `textView2`, whereas in German the text in `textView2` is longer than that in `textView1`:
+ 
+ ![Aligned Layout (German)](../assets/images/basics/barriers_alignment_de.png)
+
+The problem here is that `textView3` is still constrained to `textView1` and so `textView2` is now running in to `textView3`. This is seen most clearly in the Design view (the one with the white background).
+ 
+ The obvious solutions to this would be to either use `TableLayout`, or wrap `textView1` & `textView2` inside a vertical `LinearLayout` with `android:layout_width="wrap_content"` nd then constrain `textView3` to the end of this `LinearLayout`. But there is a better way: Barriers.
```
Comment (welsinga) [inline basics/barriers.md]: nd?

File: basics/barriers.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+ ![Create Barrier](../assets/images/basics/barrier_create.gif)
+
+You can see the `Barrier` that is added to the component tree (the lower left panel).
+ 
+We can optionally change its position within the layout hierarchy by dragging it within the component tree:
+
+ ![Move Barrier](../assets/images/basics/barrier_order.gif)
+ 
+ Next we need to set the direction fo the `Barrier`. In our case we want to position the `Barrier` relative to the `End` of either `textView1` or `textView2` depending on whichever is the larger, so we need to specify a direction of `end`:
```
Comment (welsinga) [inline basics/barriers.md]: fo  should be to

File: basics/barriers.md (hunk 2 above)
Comment (writtmeyer) [inline basics/barriers.md]: In addition to this "and": Shouldn't this be "..., or _to_ wrap ..." right after the first comma? Adding the "to" sounds better to me. But, being a German, I might be wrong with that.

**Comment by StylingAndroid** [comment]
All comments addressed @welsinga & @writtmeyer

**Pull request merged**